lib/kernel/src/code.erl

```diff
@@ -785,6 +834,97 @@ where_is_file(Tail, File, Path, Files) ->
             where_is_file(Tail, File)
     end.
 
+-spec get_doc(Mod) -> {ok, Res} | {error, Reason} when
+      Mod :: module(),
+      Res :: #docs_v1{},
+      Reason :: non_existing | missing | file:posix().
+get_doc(Mod) when is_atom(Mod) ->
+    case which(Mod) of
+        preloaded ->
+            Fn = filename:join([code:lib_dir(erts),"ebin",atom_to_list(Mod) ++ ".beam"]),
+            get_doc_chunk(Fn, Mod);
+        Error when is_atom(Error) ->
+            {error, Error};
+        Fn ->
+            get_doc_chunk(Fn, Mod)
+    end.
+
+get_doc_chunk(Filename, Mod) when is_atom(Mod) ->
+    case beam_lib:chunks(Filename, ["Docs"]) of
+        {error,beam_lib,{missing_chunk,_,_}} ->                
+            case get_doc_chunk(Filename, atom_to_list(Mod)) of
+                {error,missing} ->
+                    get_doc_chunk_from_ast(Filename);
+                Error ->
+                    Error
+            end;
+        {error,beam_lib,{file_error,Filename,enoent}} ->
+            get_doc_chunk(Filename, atom_to_list(Mod));
+        {ok, {Mod, [{"Docs",Bin}]}} ->
+            binary_to_term(Bin)
+    end;
+get_doc_chunk(Filename, Mod) ->
+    case filename:dirname(Filename) of
+        Filename ->
+            {error,missing};
+        Dir ->
+            ChunkFile = filename:join([Dir,"doc","chunks",Mod ++ ".chunk"]),
+            case file:read_file(ChunkFile) of
+                {ok, Bin} ->
+                    {ok, binary_to_term(Bin)};
+                {error,enoent} ->
+                    get_doc_chunk(Dir, Mod);
+                {error,Reason} ->
+                    {error,Reason}
+            end
+    end.
+
+get_doc_chunk_from_ast(Filename) ->
+    case beam_lib:chunks(Filename, [abstract_code]) of
+        {error,beam_lib,{missing_chunk,_,_}} ->
+            {error,missing};
+        {ok, {_Mod, [{abstract_code,
+                      {raw_abstract_v1, AST}}]}} ->
+            Docs = get_function_docs_from_ast(AST),
+            {ok, #docs_v1{ anno = 0, beam_language = erlang, format =  <<"text/erlang_doc">>,
```

Comment:
Should this be `?NATIVE_FORMAT` or `<<"text/erlang_doc">>`?

Reply:
fixed